### PR TITLE
Record object regression test

### PR DESCRIPTION
**Pull request by corbt**
Regression test for #12, as well as a fix for some tests that weren't as robust as they should have been.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ describe('transit', function() {
       });
 
       var roundTrip = recordTransit.fromJSON(recordTransit.toJSON(data));
-      expect(roundTrip).to.eql(data);
+      expect(roundTrip.toString()).to.eql(data.toString());
```
Comment (glenjamin) [inline test/test.js]: This doesn't seem right to me, we want to check deep equality of the objects - not their string representation.

What tests were failing without this? Do you have the latest version of `chai-immutable` installed?

**Comment by corbt** [comment]
The problem is that if we don't do this, `eql` converts both of them to JSON. the JSON representations of `Map` and `Record` are identical, so the check will pass even if it's not successfully deserializing as a `Record`.

Immutable's `toString` implementation appears to print deeply and includes the class names, so it correctly fails if there is any difference in the before/after representation.

**Comment by glenjamin** [comment]
The tests use `chai-immutable`, which uses `Immutable.is` for comparsion: https://github.com/astorije/chai-immutable/blob/master/chai-immutable.js#L101

**Comment by corbt** [comment]
Hmm, well then either I've got the tests misconfigured or chai-immutable is broken, because the following passes for me:

```js
      expect(new FooRecord()).to.eql(new Immutable.Map({a: 1, b: 2}));
```

**Comment by glenjamin** [comment]
Wow, this seems to actually be an issue in Immutable itself:

```js
var Immutable = require('immutable');
var FooRecord = Immutable.Record({ a: 1, b: 2 }, 'foo');
Immutable.is(new FooRecord(), new Immutable.Map({a: 1, b: 2}));
// => true
```

**Comment by corbt** [comment]
Haha, you want to open an issue on Immutable or shall I?

Also, should we stick with the `toString` comparisons for now or revert to the normal equality checks and trust that Immutable gets updated in a timely manner?

**Comment by glenjamin** [comment]
I've tweeted Lee to check if this is desired behaviour or not. If it is, then I think we should do the object check & the string check.

**Comment by glenjamin** [comment]
Ok, confirmed that this is intended behaviour in the current version of immutable: https://twitter.com/glenathan/status/688050710047539200

So lets go with `.eql` and string equal as the comparison we want.

**Comment by corbt** [comment]
PR updated. I can break it into 2 commits (regression test and `toString` additions) if it's a big deal.

File: test/test.js [hunk 2]
```diff
@@ -144,6 +145,7 @@ describe('transit', function() {
 
       var roundTrip = recordTransit.fromJSON(recordTransit.toJSON(data));
       expect(roundTrip).to.eql(data);
+      expect(roundTrip.toString()).to.eql(data.toString());
     });
 
     it('should serialize unspecified Record as a Map', function() {
```
Comment (glenjamin) [inline test/test.js]: Could you add the toString here as well please?

I wonder if it's worth writing a helper method / chai-plugin (inline, not as separate module) and using that for every comparison.

**Comment by corbt** [comment]
I've updated the PR. Added toString check to 'should serialize unspecified records as maps' and added a helper function.

**Comment by glenjamin** [comment]
Great stuff

**Comment by glenjamin** [comment]
Released as `v0.5.2`

**Comment by corbt** [comment]
:tada:

**Pull request merged**